Client/Frontend/Sync/BraveCore/Bookmarks/BookmarkAPIExtensions.swift

```diff
@@ -0,0 +1,255 @@
+    }
+    
+    func lastFolderPath() -> [BookmarkNode] {
+        if Preferences.General.showLastVisitedBookmarksFolder.value,
+           let nodeId = Preferences.Chromium.lastBookmarksFolderNodeId.value,
+           var folderNode = getNodeById(nodeId),
+           folderNode.isVisible {
+            
+            // We don't ever display the root node
+            // It is the mother of all nodes
+            let rootNodeGuid = rootNode?.guid
+            
+            var nodes = [BookmarkNode]()
+            nodes.append(folderNode)
+            
+            while true {
+                if let parent = folderNode.parent, parent.isVisible, parent.guid != rootNodeGuid {
+                    nodes.append(parent)
+                    folderNode = parent
+                    continue
+                }
+                break
+            }
+            return nodes.map({ $0 }).reversed()
+        }
+        
+        // Default folder is the mobile node..
+        if let mobileNode = mobileNode {
+            return [mobileNode]
+        }
+        
+        return []
+    }
+    
+    func mobileNode() -> BookmarkNode? {
+        if let node = mobileNode {
+            return node
+        }
+        return nil
+    }
+    
+    func fetchParent(_ bookmarkItem: BookmarkNode?) -> BookmarkNode? {
+        guard let bookmarkItem = bookmarkItem else {
+            return nil
+        }
+        
+        if let parent = bookmarkItem.parent {
+            // Return nil if the parent is the ROOT node
+            // because AddEditBookmarkTableViewController.sortFolders
+            // sorts root folders by having a nil parent.
+            // If that code changes, we should change here to match.
+            if bookmarkItem.parent?.guid != rootNode?.guid {
+                return parent
+            }
+        }
+        return nil
+    }
+    
+    @discardableResult
+    func addFolder(title: String, parentFolder: BookmarkNode? = nil) -> BookmarkNode? {
+        if let parentFolder = parentFolder {
+            return  createFolder(withParent: parentFolder, title: title)
+        } else {
+            return createFolder(withTitle: title)
+        }
+    }
+    
+    func add(url: URL, title: String?, parentFolder: BookmarkNode? = nil) {
+        if let parentFolder = parentFolder {
+            createBookmark(withParent: parentFolder, title: title ?? "", with: url)
+        } else {
+            createBookmark(withTitle: title ?? "", url: url)
+        }
+    }
+    
+    func frc(parent: BookmarkNode?) -> BookmarksV2FetchResultsController? {
+        return Bookmarkv2Fetcher(parent, api: self)
+    }
+    
+    func foldersFrc(excludedFolder: BookmarkNode? = nil) -> BookmarksV2FetchResultsController? {
+        return Bookmarkv2ExclusiveFetcher(excludedFolder, api: self)
+    }
+    
+    func getChildren(forFolder folder: BookmarkNode, includeFolders: Bool) -> [BookmarkNode]? {
+        let result = folder.children.map({ $0 })
```

Comment:
This map is not needed anymore

Reply:
Fix at https://github.com/brave/brave-ios/pull/4024/commits/afa0c0001e63c7804bd48cfdf71bfc66d730557b